src/game_destiny.cpp

```diff
@@ -0,0 +1,342 @@
+/*
+ * This file is part of EasyRPG Player.
+ *
+ * EasyRPG Player is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * EasyRPG Player is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with EasyRPG Player. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+// Headers
+#include "game_destiny.h"
+
+#ifndef EMSCRIPTEN
+#include "exe_reader.h"
+#endif // !EMSCRIPTEN
+#include "output.h"
+
+using Destiny::InterpretFlag;
+using Destiny::MainFunctions::Interpreter;
+
+using lcf::ToString;
+using lcf::rpg::EventCommand;
+using lcf::rpg::SaveEventExecFrame;
+
+
+// =-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
+//				Game_Destiny implementations
+// =-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
+
+void Game_Destiny::Load()
+{
+	// Do not load Destiny whether player cannot find "Destiny.dll"
+	if (! FileFinder::Game().Exists(DESTINY_DLL))
+	{
+		return;
+	}
+
+	uint32_t dllVersion = 0;
+	uint32_t language = 0;
+	uint32_t gameVersion = 0;
+	uint32_t extra = 0;
+	uint32_t dwordSize = 0;
+	uint32_t floatSize = 0;
+	uint32_t stringSize = 0;
+
+#ifndef EMSCRIPTEN
+	Filesystem_Stream::InputStream exe = FileFinder::Game().OpenFile(EXE_NAME);
+
+	if (exe)
+	{
+		exe.seekg(0x00030689, std::ios_base::beg);
+		exe.read(reinterpret_cast<char*>(&stringSize), sizeof(uint32_t));
+		exe.seekg(1, std::ios_base::cur);
+		exe.read(reinterpret_cast<char*>(&floatSize), sizeof(uint32_t));
+		exe.seekg(1, std::ios_base::cur);
+		exe.read(reinterpret_cast<char*>(&dwordSize), sizeof(uint32_t));
+		exe.seekg(1, std::ios_base::cur);
+		exe.read(reinterpret_cast<char*>(&extra), sizeof(uint32_t));
+		exe.seekg(1, std::ios_base::cur);
+		exe.read(reinterpret_cast<char*>(&gameVersion), sizeof(uint32_t));
+		exe.seekg(1, std::ios_base::cur);
+		exe.read(reinterpret_cast<char*>(&language), sizeof(uint32_t));
+		exe.seekg(1, std::ios_base::cur);
+		exe.read(reinterpret_cast<char*>(&dllVersion), sizeof(uint32_t));
+		exe.seekg(1, std::ios_base::cur);
+	}
+#else
+	// TODO [XGB]: Find to manage Destiny initialization parameters on Emscripten
+	dllVersion = 0x20000;
+	language = ENGLISH;
```

Comment:
This must be ``Destiny::ENGLISH``.

Reply:
Oh, that's true. My bad.
The compiler will never know where this keyword came from if I do not specify its namespace 😅